x-pack/plugins/ml/public/embeddables/anomaly_explorer/anomaly_explorer_embeddable.tsx

```diff
@@ -0,0 +1,140 @@
+  AnomalyExplorerServices,
+} from '..';
+import type { IndexPattern } from '../../../../../../src/plugins/data/common/index_patterns';
+
+export const getDefaultExplorerChartsPanelTitle = (jobIds: JobId[]) =>
+  i18n.translate('xpack.ml.anomalyExplorerChartsEmbeddable.title', {
+    defaultMessage: 'ML anomaly explorer charts for {jobIds}',
+    values: { jobIds: jobIds.join(', ') },
+  });
+
+export type IAnomalyExplorerEmbeddable = typeof AnomalyExplorerEmbeddable;
+
+export class AnomalyExplorerEmbeddable extends Embeddable<
+  AnomalyExplorerEmbeddableInput,
+  AnomalyExplorerEmbeddableOutput
+> {
+  private node?: HTMLElement;
+  private reload$ = new Subject();
+  public readonly type: string = ANOMALY_EXPLORER_CHARTS_EMBEDDABLE_TYPE;
+
+  constructor(
+    initialInput: AnomalyExplorerEmbeddableInput,
+    public services: [CoreStart, MlDependencies, AnomalyExplorerServices],
+    parent?: IContainer
+  ) {
+    super(
+      initialInput,
+      {
+        defaultTitle: initialInput.title,
+      },
+      parent
+    );
+    this.initializeOutput(initialInput);
+  }
+
+  private async initializeOutput(initialInput: AnomalyExplorerEmbeddableInput) {
+    const { anomalyExplorerService } = this.services[2];
+    const { jobIds } = initialInput;
+
+    try {
+      const jobs = await anomalyExplorerService.getCombinedJobs(jobIds);
+      const indexPatternsService = this.services[1].data.indexPatterns;
+
+      // First get list of unique indices from the selected jobs
+      const indices = new Set<string>();
+      jobs.forEach((j) => {
+        j.datafeed_config.indices.forEach((index) => {
+          if (!indices.has(index)) {
+            indices.add(index);
+          }
+        });
+      });
+
+      // Then find the index patterns assuming the index pattern title matches the index name
+      const indexPatterns: Record<string, IndexPattern> = {};
+      for (const indexName of indices) {
+        const response = await indexPatternsService.find(`"${indexName}"`);
+
+        const ip = response.find((obj) => obj.title.toLowerCase() === indexName.toLowerCase());
+        if (ip !== undefined) {
+          indexPatterns[ip.id!] = ip;
+        }
+      }
+
+      this.updateOutput({
+        ...this.getOutput(),
+        indexPatterns: Object.values(indexPatterns),
+      });
+    } catch (e) {
+      // Unable to find and load index pattern but we can ignore the error
+      // as we only load it to support the filter & query bar
+      // the visualizations should still work correctly
+
+      // eslint-disable-next-line no-console
+      console.error(`Unable to load index patterns for ${jobIds}`, e);
+    }
+  }
+
+  public render(node: HTMLElement) {
+    super.render(node);
+    this.node = node;
+
+    const I18nContext = this.services[0].i18n.Context;
+
+    ReactDOM.render(
+      <I18nContext>
+        <KibanaContextProvider services={{ ...this.services[0] }}>
+          <Suspense fallback={null}>
```

Comment:
Do we want to provide a loading state? We do that for the Vega charts, there's an example in `/plugins/ml/public/application/components/vega_chart/vega_chart.tsx`.

Reply:
Updated here https://github.com/elastic/kibana/pull/94396/commits/3b3e468468aee5e6edef238695517df46bb7f710